README-Networking.md

```diff
@@ -1,17 +1,40 @@
 # Networking in Container Management
 
 ## Overview
+Kubernetes approaches networking somewhat differently than Docker does by default. There are 4 distinct networking problems to solve:
+
+  - Highly-coupled container-to-container communications: this is solved by pods and localhost communications.
+  - Pod-to-Pod communications
+  - Pod-to-Service communications: done using services.
+  - External-to-Service communications: done using services.
+
+Pods can be treated much like VMs or physical hosts from the perspectives of port allocation, naming, service discovery, load balancing, application configuration, and migration.  This means that you do not need to explicitly create links between pods and you almost never need to deal with mapping container ports to host ports.  Containers are addressed with routable IP addresses that are on the subnet of the external network.  
```

Comment:
Add this right before here, 

Kubernetes assumes that pods can communicate with other pods, regardless of which host they land on. Every pod gets its own IP address so you do not need to explicitly create links between pods and you almost never need to deal with mapping container ports to host ports.

Reply:
(right before Pods can be treated....)